app/javascript/components/TradeProposal/TradeProposalModal.jsx

```diff
@@ -1,79 +1,73 @@
 import React, { useState } from 'react';
 import classNames from 'classnames';
 import xhrRequest from '../../helpers/xhr-request';
+import { saveScrollPos } from '../../helpers/scroll-restoration';
+import LoadingIcon from '../Icons/LoadingIcon';
 import Button from '../Button';
 import Modal from '../Modal';
 
-const sendTradeMessage = (fromUserId, toUserId,  cardId) => xhrRequest({
-    url: '/trade_message',
-    options: {
-      method: 'POST',
-      body: JSON.stringify({ from_user_id: fromUserId, to_user_id: toUserId, card_id: cardId}),
-    },
-})
+const refreshPage = () => {
+    saveScrollPos();
+    // eslint-disable-next-line no-undef
+    Turbolinks.visit(window.location.href, { action: 'replace' });
+}
 
-const refreshPage = () => window.location.reload();
+const Status = {
+    initial: () => (<><i className="fab fa-discord" /> Send Message</>),
+    loading: () => 'Sending...',
+    success: () => 'Message Sent!',
+    error: () => (<><i className="fas fa-exclamation-circle" /> Message Error</>)
+}
 
 const TradeProposalModal = ({ onClose, card, currentUserId, user, priorMessageTimestamp }) => {
     const [reqState, setReqState] = useState('initial');
+    const isComplete = reqState === 'success' || reqState === 'error';
+    const sendMessage = async () => {
+        setReqState('loading');
+        try {
+            await xhrRequest({
+                url: '/trade_message',
+                options: {
+                  method: 'POST',
+                  body: JSON.stringify({ from_user_id: currentUserId, to_user_id: user.id, card_id: card.id}),
+            }});
+            setReqState('success');
+        } catch (e) {
+            setReqState('error');
+        }
+    };
+
     return (
-        <Modal onClose={reqState === 'initial' ? onClose : refreshPage}>
-            <div style={{ marginBottom: '1rem', display: 'flex', flexDirection: 'column', gap: '1rem' }}>
+        <Modal onClose={isComplete ? onClose : refreshPage}>
+            <div className='trade-message--prompt'>
                 <div>
-                    <div style={{ fontSize: '.8rem', fontWeight: '700'}}>You are contacting:</div>
+                    <div className='heading'>You are contacting:</div>
                     <div>{user.name}</div>
                 </div>
                 <div>
-                    <div style={{ fontSize: '.8rem', fontWeight: '700'}}>To talk about trading for:</div>
+                    <div className='heading'>To talk about trading for:</div>
```

Comment:
Not gonna really stop my approval, but should these be h tags for accessibility?  Just using the correct semantic element.

Reply:
Catching me on my lazy markup... you're right though

Reply:
Fixed